core/modules/field_ui/tests/field_ui.test

```diff
@@ -832,3 +831,291 @@ class FieldUIAlterTestCase extends BackdropWebTestCase {
+    $this->assertViewModeExists('node', 'custom_1');
+    $this->assertViewModeDefaultDisplay('node', 'post', 'custom_1');
+    $this->assertViewModeDefaultDisplay('node', 'page', 'custom_1');
+
+    // Enable the view-mode on the page content type.
+    $this->backdropGet('admin/structure/types/manage/page/display/view-modes/custom_1/enable');
+    $this->assertViewModeExists('node', 'custom_1');
+    $this->assertViewModeCustomDisplay('node', 'page', 'custom_1');
+    $this->assertViewModeDefaultDisplay('node', 'post', 'custom_1');
+
+    // Test that the default settings are applied to newly enabled view modes.
+    $settings = field_bundle_settings('node', 'page');
+    $this->assertEqual($settings['extra_fields']['display']['test']['custom_1']['testing'], FALSE, 'The default settings were applied to the newly enabled view mode.');
+
+    // Delete the view mode.
+    $this->backdropPost('admin/structure/types/manage/page/display/view-modes/custom_1/delete', array(), 'Delete');
+    $this->assertText('Deleted the Custom 1 view mode.');
+    $this->assertNoViewModeExists('node', 'custom_1');
+    $this->assertViewModeDefaultDisplay('node', 'page', 'custom_1');
+    $this->assertViewModeDefaultDisplay('node', 'post', 'custom_1');
+  }
+
+  /**
+   * Test the new entity view mode hooks.
+   */
+  public function testInfoHooks() {
+    config_set('entity.view_modes', 'view_modes', array(
+      'node' => array(
+        'info_3' => array(
+          'label' => t('Variable-altered view mode'),
+          'custom settings' => TRUE,
+        ),
+      ),
+      'taxonomy_term' => array(),
+    ));
+    $this->refreshVariables();
+
+    $info = entity_get_info();
+
+    // An invalid entity type in hook_entity_view_mode_info() does not pass
+    // into the entity info array.
+    $this->assertTrue(!isset($info['invalid-type']));
+
+    // Test hook-provided view modes.
+    $this->assertIdentical($info['node']['view modes']['info_1'], array(
+      'label' => t('Hook-defined view mode #1'),
+      'custom settings' => FALSE,
+    ));
+    $this->assertIdentical($info['node']['view modes']['info_2'], array(
+      'label' => t('Hook-altered view mode'),
+      'custom settings' => TRUE,
+    ));
+    $this->assertIdentical($info['node']['view modes']['info_3'], array(
+      'label' => t('Variable-altered view mode'),
+      'custom settings' => TRUE,
+    ));
+    $this->assertIdentical($info['taxonomy_term']['view modes']['info_1'], array(
+      'label' => t('Hook-defined view mode #1'),
+      'custom settings' => TRUE,
+    ));
+
+    // Test that entity view modes defined in hook_entity_info() are never
+    // overridden by custom view modes.
+    $this->assertIdentical($info['node']['view modes']['full'], array(
+      'label' => 'Full content',
+      'custom settings' => FALSE,
+    ));
+  }
+}
+
+/**
+ * Tests view mode altering theme suggestions.
+ */
+class FieldUIViewModeSuggestionsTest extends FieldUIViewModeTestHelper {
```

Comment:
oi, those tests completely failed without some reworking to get all the theme hook suggestions in place.